### PR TITLE
Minor ProjectCrackerTool refactoring

### PR DESCRIPTION
**Pull request by ncave**
Moving ProjectCrackerTool to its own module to enable reuse.

### CODE REVIEW

**Comment by 7sharp9** [comment]
I didn't realise there was as much to do on fcs, is there much left?

**Comment by dsyme** [comment]
@7sharp9 This is for the .NET Core 1.0 (i.e. .NET Standard 1.5) build of FCS

**Comment by dsyme** [comment]
This looks good to me. @rneatherway if you get a chance could you take a look?

File: src/fsharp/FSharp.Compiler.Service.ProjectCrackerTool/ProjectCrackerOptions.fs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ﻿namespace Microsoft.FSharp.Compiler.SourceCodeServices.ProjectCrackerTool
 
+[<CLIMutable>]
```
Comment (dsyme) [inline src/fsharp/FSharp.Compiler.Service.ProjectCrackerTool/ProjectCrackerOptions.fs]: Why is CLIMutable needed? thanks

**Comment by ncave** [comment]
@dsyme  [<CLIMutable>] is needed for serialization on .NET Core.

**Comment by 7sharp9** [comment]
Yeah I know, I wondered how much was left on that. Having a full independent .netcore compiler and tools would be interesting. E.g having no mono installed at all.

**Comment by rneatherway** [comment]
Hi guys. I'm not really up to speed on .NET Core, but this PR seems sensible and behaviour-preserving.

**Pull request merged**